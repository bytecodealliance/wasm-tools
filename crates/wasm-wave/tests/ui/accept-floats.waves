list-float32([0, 0.0, 0e0, 0.0e0, 0e-1, 0e+1, 0.000e100]);
list-float32([-3.1415, -100000, -0.0e-0]);
list-float32([nan, inf, -inf]);
// Largest normal f32
float32(3.4028234664e38);
// Truncated precision
float32(3.4028234664123e38);
// Too large; infinity
float32(3.4028234664e39);
// Smallest positive non-zero f32
float32(1.4012984643e-45);
// Too small; round to zero
float32(1.4012984643e-46);