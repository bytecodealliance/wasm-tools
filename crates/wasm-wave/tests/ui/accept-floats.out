list-float32(vals: [0, 0, 0, 0, 0, 0, 0])
list-float32(vals: [-3.1415, -100000, -0])
list-float32(vals: [nan, inf, -inf])
// Largest normal f32
float32(val: 340282350000000000000000000000000000000)
// Truncated precision
float32(val: 340282350000000000000000000000000000000)
// Too large; infinity
float32(val: inf)
// Smallest positive non-zero f32
float32(val: 0.000000000000000000000000000000000000000000001)
// Too small; round to zero
float32(val: 0)
